src/core/config/config.rs

```diff
@@ -604,6 +608,30 @@ pub struct Call {
     pub steps: Vec<Step>,
 }
 
+///
+/// Provides the ability to load custom extensions that are executed before the
+/// execution of the IR for the field and after the field value is resolved.
+#[derive(
+    Serialize,
+    Deserialize,
+    Clone,
+    Debug,
+    Default,
+    PartialEq,
+    Eq,
+    schemars::JsonSchema,
+    DirectiveDefinition,
+)]
+#[directive_definition(locations = "FieldDefinition")]
+pub struct Extension {
+    /// Used to define where the extension is located so tailcall can load it.
+    pub name: String,
+
+    /// Used to define parameters that are calculated and injected into the IR.
+    /// Mustache syntax can be used on the parameters field.
+    pub params: Value,
```

Comment:
This should be a `
```suggestion
    pub params: BTreeMap<String, Value>,
```

Reply:
Value is JSONLike. It can act as a superset and accept Maps.
It can accept `["test", "{{.args.test}}", 1]`, `false`, `{ "hello": "world", "foo": 12, "bar": true }`, etc. 

Do we need to force a `{ [Key]: [Value] }` structure? or should we leave the more flexible format?

If we change the format I have to change the function signatures from `JsonLike` to `BTreeMap<String, JsonLike>`